src/coreclr/gc/gcimpl.h

```diff
@@ -202,6 +202,8 @@ class GCHeap : public IGCHeapInternal
 
     int StartNoGCRegion(uint64_t totalSize, bool lohSizeKnown, uint64_t lohSize, bool disallowFullBlockingGC);
     int EndNoGCRegion();
+    enable_no_gc_region_callback_status EnableNoGCRegionCallback(uint64_t callback_threshold);
+    finalizer_action_kind GetNoGCFinalizerAction(bool reset);
```

Comment:
`bool reset` flag should not be needed anymore.

Reply:
Nope, we need to make sure the call at `Thread::HaveExtraWorkForFinalizer` does not reset the flag.

Reply:
You can refactor `HaveExtraWorkForFinalizer`, or maybe even get rid `HaveExtraWorkForFinalizer`, to make this flag unnecessary.

The initial intention with `HaveExtraWorkForFinalizer` was to make it a super-fast inlineable check. This intention got lost over time as people kept adding more stuff to it, and this change makes it even worse by adding a virtual call over JIT/EE interface to it.

Reply:
> The GC team is working under the constraint that the new GC should work with old runtimes. The preferred approach is by only introducing new APIs on IGCHeap. 

This is going to lead to convoluted interface over time. I think it is a bad approach for the long run.

I think you should be adding the new methods to the GC/EE interface as if this constraint did not exist, and then wrap them in the versioning checks as necessary for back-compat. Note that these versioning checks are temporary. Over time, the oldest version checks can be deleted as older runtime versions go out of support.

What would be your preferred design of the GC<->EE intefaces wihout taking the versioning constraint into account?

Reply:
Ok. Here is the cleanup that I had mind in the original comment:

https://github.com/jkotas/runtime/commit/ba10a0ebf100863b8afab583f365f3cbb9a03eb9

Does that work? (Feel free to cherry-pick the commit into this PR.)

Reply:
Thanks, this change looks interesting. I am testing it to see if that works!